app/waf/src/services/bigip.service.ts

```diff
@@ -97,6 +97,26 @@ export class BigIpManager {
     );
   }
 
+  async getLicense(): Promise<BigipLicense> {
+    await this.mustBeReachable();
+
+    let url = `${this.baseUrl}/mgmt/tm/sys/license`;
+
+    let response = await this.bigipService.getInfo(url, this.cred64Encoded);
+    let resObj = JSON.parse(JSON.stringify(response))['body'][0];
+    this.logger.debug(`get ${url} resposes: ${JSON.stringify(resObj)}`);
+
+    for (let entry of Object.keys(resObj.entries)) {
+      if (!entry.endsWith('/mgmt/tm/sys/license/0')) continue;
```

Comment:
This is not bad, it sounds useful in the future :D

Reply:
Yeah. However, there may still be some adjustment in the future to adopt the coming uncertainties.